test/dummy.go

```diff
@@ -2,18 +2,19 @@ package test
 
 import (
 	"bytes"
-
 	"context"
 	"crypto/sha512"
 	"fmt"
 	"slices"
+	"sync"
 	"time"
 
 	"github.com/rollkit/go-execution/types"
 )
 
 // DummyExecutor is a dummy implementation of the DummyExecutor interface for testing
 type DummyExecutor struct {
+	mu           sync.RWMutex // Add mutex for thread safety
```

Comment:
since `stateRoot` can be read without locks, its better to provide a mutex read method and update the place where it is directly accessing `stateRoot` to use this mutex method.

```
func (e *DummyExecutor) GetStateRoot() types.Hash {
	e.mu.RLock()
	defer e.mu.RUnlock()

	return e.stateRoot
}
```